include/boost/math/special_functions/detail/ibeta_inverse.hpp

```diff
@@ -673,7 +673,34 @@ T ibeta_inv_imp(T a, T b, T p, T q, const Policy& pol, T* py)
          invert = !invert;
          xs = 1 - xs;
       }
-      T xg = pow(a * p * boost::math::beta(a, b, pol), 1/a);
+      if (a < tools::min_value<T>())
+      {
+         if (py)
+         {
+            *py = invert ? 0 : 1;
+         }
+         return invert ? 1 : 0; // nothing interesting going on here.
+      }
+      //
+      // The call to beta may overflow, plus the alternative using lgamma may do the same
+      // if T is a type where 1/T is infinite for small values (denorms for example).
+      //
+      T bet = 0;
+      T xg;
+      bool overflow = false;
+      try {
+         bet = boost::math::beta(a, b, pol);
+      }
+      catch (const std::runtime_error&)
+      {
+         overflow = true;
+      }
+      if (overflow || !(boost::math::isfinite)(bet))
+      {
+         xg = exp((boost::math::lgamma(a + 1, pol) + boost::math::lgamma(b + 1, pol) - boost::math::lgamma(a + b + 1, pol) + log(p) - log(b) + log(a + b)) / a);
```

Comment:
Maybe I'm missing something, but why not reduce

`lgamma(b+1) - log(b) - lgamma(a+b+1) + log(a+b)`

To just `lgamma(b) - lgamma(a+b)`

?

Reply:
The branch deals with a,b < 1 and the problem occurs when one of a,b is so small that 1/a (for example) overflows, so I had issues with lgamma overflowing as well.... though in point of fact I guess that may be a bug as well :(

Reply:
It was a bug in lgamma: there was actually a fix for this already in place, it just didn't quite go far enough.